File: javapatterns/src/main/java/org/iut/nantes/state/car/Etat.java
package org.iut.nantes.state.car;

abstract class Etat
{
  protected GoogleCar car;

  public Etat( GoogleCar gc )
  {
    car = gc;
  }

  public void bouton() {}
  public void saisieDestination() {}
  public void arrive() {}
}

File: kotlinpatterns/src/main/kotlin/org/iut/nantes/state/distributeur/etats/Etat.kt
package org.iut.nantes.state.distributeur.etats

import org.iut.nantes.state.distributeur.Distributeur

abstract class Etat(distributeur: Distributeur) {
    abstract fun message()

    open fun insererPiece() {}
    open fun annuler() {}
    open fun tournerPoignee() {}

}